crates/subspace-service/src/sync_from_dsn/snap_sync.rs

```diff
@@ -241,6 +239,12 @@ where
 }
 
 #[allow(clippy::too_many_arguments)]
+/// Synchronize the blockchain to the target_block (approximate value based on the containing
+/// segment) or to the last archived block.
+///
+/// Note: setting import_state_block_only will import only the block related to the state (a block
+/// number equal or less than target_block or the first block of the last archived segment) and
+/// disable importing the remaining blocks of the downloaded segment.
```

Comment:
I thought `target_block` is target block, but this comment is saying a lower block number may be imported instead, is that really true and if so then why not importing the block that is requested as one would expect?

Reply:
Our current code supports the case of importing the first block of the segment with a state and possibly importing the remaining block later. Are suggesting to implement: 1) import blocks before the block, 2) import the state block, 3) import the remaining blocks ?

Reply:
I'm not suggesting anything, I'm asking about what was actually implemented because the comment doesn't match what I expected the code would do.